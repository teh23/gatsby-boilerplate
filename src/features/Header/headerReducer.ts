import { createSlice, PayloadAction } from '@reduxjs/toolkit'

export type MenuList = string[]

const initialState: {
  menuList: MenuList
} = {
  menuList: ['Main page', 'Products', 'Contact'],
}

const headerSlice = createSlice({
  name: 'header',
  initialState,
  reducers: {
    hideProducts: (state, action: PayloadAction<boolean>) => {
      const { payload } = action
      if (payload === true) {
        state.menuList = state.menuList.filter((item) => item !== 'Products')
      } else {
        state.menuList = initialState.menuList
      }
    },
  },
})

// Reducers
export default headerSlice.reducer

// Actions
const { hideProducts } = headerSlice.actions
