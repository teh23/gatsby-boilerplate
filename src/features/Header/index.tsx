import React from 'react'
import { useAppDispatch, useAppSelector } from '../../helpers/hooks/redux'

const Header: React.FunctionComponent<{
  title: string
}> = ({ title }): JSX.Element => {
  const menuList = useAppSelector((state) => state.header.menuList)
  // const dispatch = useAppDispatch()

  return (
    <header className="bg-gray-600">
      <div className="container flex justify-between items-center mx-auto h-16">
        <div id="logo">
          <p className="font-bold text-lg">{title}</p>
        </div>
        <div id="menu">
          <ul className="flex space-x-4">
            {menuList.map((item: string) => (
              <li key={item}>{item}</li>
            ))}
          </ul>
        </div>
      </div>
    </header>
  )
}

export default Header
