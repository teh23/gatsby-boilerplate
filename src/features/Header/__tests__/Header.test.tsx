import React from 'react'
import renderer from 'react-test-renderer'
import Header from '..'

describe('Header', () => {
  it('first', () => {
    const Test = renderer.create(<Header title="asd" />).toJSON()
    expect(Test).toMatchSnapshot()
  })
})
