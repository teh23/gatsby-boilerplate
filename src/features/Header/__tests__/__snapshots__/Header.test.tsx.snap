// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header first 1`] = `
<div
  className="h-24 bg-gray-600"
>
  asd
</div>
`;
